backend/src/test/java/com/github/hamideh6182/service/BookServiceTest.java

```diff
@@ -163,4 +185,45 @@ void increaseBookQuantityTest_WhenQuantityPlusOne() {
         verify(bookRepository).save(book2);
         assertEquals(expected, actual);
     }
+
+    @Test
+    void increaseBookQuantityTest_whenBookDoesntExist_thenThrowException() {
+        //WHEN
+        when(bookRepository.findById("4")).thenReturn(Optional.empty());
+        //THEN
+        assertThrows(BookNotFoundException.class, () -> bookService.increaseBookQuantity("4"));
+        verify(bookRepository).findById("4");
+    }
+
+    @Test
+    void decreaseBookQuantityTest_WhenQuantityMinusOne() {
```

Comment:
maybe you should here name the test something like: when_decreaseBookQuantity_then_Quantity_is_MinusOne